alfi/solver.py

```diff
@@ -616,7 +661,7 @@ def residual(self):
         F = (
             self.nu * inner(2*sym(grad(u)), grad(v))*dx
             + self.gamma * inner(div(u), div(v))*dx
-            + self.advect * inner(dot(grad(u), u), v)*dx
+            + self.advect * inner(dot(grad(u), self.beta), v)*dx
```

Comment:
This is just adapting to possibly having a wind (for fully nonlinear, beta is u).